### PR TITLE
Bluetooth: Mesh: Change the default value to Kconfig

### PR DESCRIPTION
**Pull request by LingaoM**
Move default value of segment retransmit and timeout value
to Kconfig.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/Kconfig [hunk 1]
```diff
@@ -292,6 +299,21 @@ config BT_MESH_TX_SEG_MAX
 	  which leaves 56 bytes for application layer data using a
 	  4-byte MIC and 52 bytes using an 8-byte MIC.
 
+config BT_MESH_TX_SEG_RETRANS_CNT
+	int "Number of retransmit attempts"
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:303]: This ends up in the same menu as the rest of the mesh configuration, could you specify that this is for segmented transport messages? E.g. "Transport message segment retransmit attempts"

File: subsys/bluetooth/mesh/Kconfig [hunk 2]
```diff
@@ -292,6 +299,21 @@ config BT_MESH_TX_SEG_MAX
...
+	default 4
+	range 1 8
+	help
+	  Maximun number of retransmit attempts for outgoing segment
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:307]:
```suggestion
	  Maximum number of retransmit attempts for outgoing segment 
```

File: subsys/bluetooth/mesh/Kconfig [hunk 3]
```diff
@@ -292,6 +299,21 @@ config BT_MESH_TX_SEG_MAX
...
+config BT_MESH_TX_SEG_RETRANS_CNT
+	int "Number of retransmit attempts"
+	default 4
+	range 1 8
+	help
+	  Maximun number of retransmit attempts for outgoing segment 
+	  message.
+
+config BT_MESH_TX_SEG_RETRANS_TIMEOUT
+	int "Timeout value of retransmit segment message to group address"
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:311]: Same as above, e.g. "Transport message segment retransmit interval for group messages"

File: subsys/bluetooth/mesh/transport.c [hunk 4]
```diff
@@ -63,7 +63,7 @@
  * reason to delay the repetitions significantly. Delaying by more than 0 ms
  * to avoid flooding the network.
  */
-#define SEG_RETRANSMIT_TIMEOUT_GROUP 50
+#define SEG_RETRANSMIT_TIMEOUT_GROUP CONFIG_BT_MESH_TX_SEG_RETRANS_TIMEOUT
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/transport.c:66]: The unicast timeout components should probably be tuneable as well

File: subsys/bluetooth/mesh/pb_adv.c [hunk 5]
```diff
@@ -46,7 +46,7 @@
 #define XACT_ID_MAX  0x7f
 #define XACT_ID_NVAL 0xff
 
-#define RETRANSMIT_TIMEOUT  K_MSEC(500)
+#define RETRANSMIT_TIMEOUT  K_MSEC(CONFIG_BT_MESH_PB_ADV_RETRANS_TIMEOUT)
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/pb_adv.c:49]: This looks like a bugfix for a separate issue. Could you put this in a separate commit? Can go in the same PR.

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/pb_adv.c (hunk 5 above)
Comment (LingaoM) [reply subsys/bluetooth/mesh/pb_adv.c:49]: ooh, i'm not see any bugfix for this? please point to me.

Comment (trond-snekvik) [reply subsys/bluetooth/mesh/pb_adv.c:49]: I mean, this is fixing a separate issue from the transport.c segmentation timing, so it should be a separate commit

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/transport.c [hunk 6]
```diff
@@ -52,18 +52,21 @@
...
+#define SEG_RETRANSMIT_TIMEOUT_UNICAST(tx) \
+	(CONFIG_BT_MESH_TX_SEG_RETRANS_TIMEOUT_UNICAST + 50 * (tx)->ttl)
+
 /* When sending to a group, the messages are not acknowledged, and there's no
  * reason to delay the repetitions significantly. Delaying by more than 0 ms
  * to avoid flooding the network.
  */
-#define SEG_RETRANSMIT_TIMEOUT_GROUP 50
+#define SEG_RETRANSMIT_TIMEOUT_GROUP \
+	(CONFIG_BT_MESH_TX_SEG_RETRANS_TIMEOUT_GROUP + 50 * (tx)->ttl)
```
[the remark this answers is not in the document]
Comment (LingaoM) [reply subsys/bluetooth/mesh/transport.c]: Well, My original intention is that the time algorithm based on TTL can reduce network flooding when send multi-segment message to group address. Indeed, but It indeed increase the time consume to send multi-segment message to group address.

**Review by trond-snekvik — APPROVED** [review]

File: subsys/bluetooth/mesh/Kconfig [hunk 7]
```diff
@@ -316,6 +323,28 @@ config BT_MESH_LOOPBACK_BUFS
...
+	help
+	  Maximum number of transport message segment retransmit attempts
+	  for outgoing segment message.
+
+config BT_MESH_TX_SEG_RETRANS_TIMEOUT_UNICAST
+	int "Transport message segment retransmit interval for unicast messages"
+	default 400
+	range 200 500
+	help
+	  Maximum timeout value of retransmit segment message to unicast address.
```
Comment (joerchan) [inline subsys/bluetooth/mesh/Kconfig:339]: A timeout is the "Maximum time", but a "maximum timeout" would make sense if the timeout was negotiable.
Since it is the actual timeout I would drop the word "maximum" here.

**Review by joerchan — APPROVED** [review]

File: subsys/bluetooth/mesh/Kconfig [hunk 8]
```diff
@@ -38,6 +38,13 @@ config BT_MESH_UNPROV_BEACON_INT
 	  This option specifies the interval (in seconds) at which the
 	  device sends unprovisioned beacon.
 
+config BT_MESH_PB_ADV_RETRANS_TIMEOUT
+	int "Timeout value of retransmit provisioning pdu"
+	default 500
+	range 100 800
+	help
+	  Maximum timeout value of retransmit provisioning pdu.
```
Comment (joerchan) [inline subsys/bluetooth/mesh/Kconfig:46]:
```suggestion
	  Timeout value of retransmit provisioning pdu.
```

**Comment by carlescufi** [comment]
@jhedberg could you review?

**Review by carlescufi — APPROVED** [review]
LGTM.

File: subsys/bluetooth/mesh/Kconfig [hunk 9]
```diff
@@ -38,6 +38,13 @@ config BT_MESH_UNPROV_BEACON_INT
 	  This option specifies the interval (in seconds) at which the
 	  device sends unprovisioned beacon.
 
+config BT_MESH_PB_ADV_RETRANS_TIMEOUT
+	int "Timeout value of retransmit provisioning pdu"
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/Kconfig:42]: minor: capitalize PDU

File: subsys/bluetooth/mesh/Kconfig [hunk 10]
```diff
@@ -38,6 +38,13 @@ config BT_MESH_UNPROV_BEACON_INT
...
+	default 500
+	range 100 800
+	help
+	  Timeout value of retransmit provisioning pdu.
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/Kconfig:46]: same here

File: subsys/bluetooth/mesh/Kconfig [hunk 11]
```diff
@@ -316,6 +323,28 @@ config BT_MESH_LOOPBACK_BUFS
 	  The number of buffers allocated for the network loopback mechanism.
 	  Loopback is used when the device sends messages to itself.
 
+config BT_MESH_TX_SEG_RETRANS_CNT
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/Kconfig:326]: Sortening "COUNT" to "CNT" doesn't really do much here. The variable would anyway not be the longest one that you're introducing, so just spell out COUNT.

**Review by jhedberg — CHANGES_REQUESTED** [review]
A couple of minor naming related comments. Otherwise looks alright to me.

**Review by jhedberg — APPROVED** [review]
